node-services/sharing-service/controllers/sharedWithOthers.js

```diff
@@ -0,0 +1,63 @@
+const db = require('../dbInstance');
+const { validateEmail, validateUserParameter } = require('./validateParameters');
+
+// checkIfExists responseCodes
+// 0 -> Report not shared beforehand
+// 1 -> Report already shared
+
+const dbOperations = {
+  checkIfExists : async (reportObject) => {
+    if (!validateEmail(reportObject.email)) {
+      throw new Error("Invalid email");
+    }
+    if (!validateUserParameter(reportObject.id)) {
+      throw new Error("Invalid parameter");
+    }
+    const collectionReference = db.collection('sharedWithOthers');
+    const reportSnapshot = await collectionReference
+        .where('reportId', '==', reportObject.id).get();
+    if (reportSnapshot.empty) {
+      const docRef = await collectionReference.add({
+        reportId: reportObject.id
+      });
+      return { responseCode: '0' }
+    }
+    let docId;
+    reportSnapshot.forEach(doc => {
+      docId = doc.id;
+    });
+    const userSnapshot = await collectionReference.doc(docId)
+        .collection('users').where('email', '==', reportObject.email).get();
+    if (userSnapshot.empty) {
+      return { responseCode: '0' }
+    }
+    return { responseCode: '1' }
+  }
+  ,shareWithOthers : async (reportObject) => {
+    const collectionReference = db.collection('sharedWithOthers');
+    const reportSnapshot = await collectionReference
+        .where('reportId', '==', reportObject.id).get();
+    let docId;
+    reportSnapshot.forEach(doc => {
+      docId = doc.id;
+    });
+    const maxTries = 5;
+    tryRequest = async (currentAttempt, delay) => {
+      try {
+        const userRef = await collectionReference.doc(docId).collection('users').add({
+            email: reportObject.email
```

Comment:
why does reportObject has email of person you are sharing it with? If it is then it should not be called reportObject it should be some variation of request object. Or you can take two parameter (report, targetUser/userToShareWith)